# HG changeset patch
# User Rik <rik@octave.org>
# Date 1429414739 25200
#      Sat Apr 18 20:38:59 2015 -0700
# Node ID 0b83a442ff7835af08c7620ce735eea075d9304a
# Parent  ce027487af26e83160196a15f4c40c7ee215c426
Process SHIFT/CTRL/ALT keys properly for ginput (Bug #44833).

* ginput.m: Return the Character field (the combined keystroke) rather
than the Key field.

diff --git a/scripts/plot/util/ginput.m b/scripts/plot/util/ginput.m
--- a/scripts/plot/util/ginput.m
+++ b/scripts/plot/util/ginput.m
@@ -125,20 +125,23 @@ endfunction
 
 function ginput_windowbuttondownfcn (src, button)
   point = get (gca (), "currentpoint");
   ginput_accumulator (1, point(1,1), point(1,2), button);
 endfunction
 
 function ginput_keypressfcn (src, evt)
   point = get (gca (), "currentpoint");
-  key = evt.Key;
-  if (key == "return")
+  if (strcmp (evt.Key, "return"))
     ## Enter key stops ginput.
     ginput_accumulator (2, NaN, NaN, NaN);
   else
-    ginput_accumulator (1, point(1,1), point(1,2), uint8 (key(1)));
+    character = evt.Character;
+    if (! isempty (character))
+      ginput_accumulator (1, point(1,1), point(1,2), uint8 (character(1)));
+    endif
   endif
 endfunction
 
+
 ## Remove from test statistics.  No real tests possible.
 %!test
 %! assert (1);
